src/librustc_typeck/check/mod.rs

```diff
@@ -5084,8 +5080,36 @@ pub fn check_intrinsic_type(ccx: &CrateCtxt, it: &ast::ForeignItem) {
             "discriminant_value" => (1, vec![
                     tcx.mk_imm_ref(tcx.mk_region(ty::ReLateBound(ty::DebruijnIndex::new(1),
                                                                   ty::BrAnon(0))),
-                                    param(ccx, 0))], tcx.types.u64),
-
+                                   param(ccx, 0))], tcx.types.u64),
+            "simd_eq" | "simd_ne" | "simd_lt" | "simd_le" | "simd_gt" | "simd_ge" => {
+                (2, vec![param(ccx, 0), param(ccx, 0)], param(ccx, 1))
+            }
+            "simd_insert" => (2, vec![param(ccx, 0), tcx.types.u32, param(ccx, 1)], param(ccx, 0)),
+            "simd_extract" => (2, vec![param(ccx, 0), tcx.types.u32], param(ccx, 1)),
+            name if name.starts_with("simd_shuffle") => {
+                match name["simd_shuffle".len()..].parse() {
+                    Ok(n) => {
+                        let mut params = vec![param(ccx, 0), param(ccx, 0)];
+                        params.extend(iter::repeat(tcx.types.u32).take(n));
+
+                        let ictxt = infer::new_infer_ctxt(tcx, &tcx.tables, None, false);
+                        let ret = ictxt.next_ty_var();
+                        infer_ctxt = Some(ictxt);
+                        (2, params, ret)
+                    }
+                    Err(_) => {
+                        span_err!(tcx.sess, it.span, E0439,
+                                  "invalid `simd_shuffle`, needs length: `{}`", name);
+                        return
+                    }
+                }
+            }
+            name if name.starts_with("x86_") ||
+                    name.starts_with("arm_") ||
+                    name.starts_with("aarch64_") => {
+                // FIXME: skip checking these for now
```

Comment:
Would this be easier to wire up if the `Ty` representation was used already for the intrinsic definitions?